### ISSUE TITLE
Stopping and re-starting crawler always fails

### ISSUE THREAD

**Issue by tepez** [cgiffard/node-simplecrawler#330]
The following code:
```
const Crawler = require('simplecrawler');

const crawler = new Crawler("http://www.example.com/");
crawler.start();
crawler.stop();
crawler.start();
```
Always throws:
```
Error: Resource already exists in queue!
    at \path\to\node_modules\simplecrawler\lib\queue.js:144:25
    at FetchQueue.exists (\path\to\node_modules\simplecrawler\lib\queue.js:166:9)
    at FetchQueue.add (\path\to\node_modules\simplecrawler\lib\queue.js:132:11)
    at Crawler.start (\path\to\node_modules\simplecrawler\lib\crawler.js:440:19)
    at Object.<anonymous> (\path\to\bug.js:6:9)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Function.Module.runMain (module.js:441:10)
```
This is because `.start()` always [adds](https://github.com/cgiffard/node-simplecrawler/blob/master/lib/crawler.js#L440) a queue item based on the initial url.

This is good for *starting* a crawl, but not when *resuming* a crawl.

**Comment by cgiffard** [comment]
Good catch. It should only do that if the queue length is zero.

**Comment by fredrikekelund** [comment]
Thanks again for reporting this @tepez! I've submitted a patch that makes Crawler#start tolerate duplicate errors from the queue. We changed the behavior of this method in the cycle leading up to 1.0.0 so that it would add the initial URL to the queue even if the queue length wasn't 0, since users might want to add URL's to the queue before initiating the crawler, and they'd still expect the initial URL to be added to the queue as well. We didn't consider the restart scenario, though, and we didn't have any tests for it either. That's all fixed now!
